public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+  const [delayValue, setDelayValue] = useState(1);
+  const [enabled, setEnabled] = useState<boolean>(false);
+  const [shingleSizeValue, setShingleSizeValue] = useState(8);
+
+  const title = embeddable.getTitle();
+  const onAccordionToggle = (key) => {
+    const newAccordionsOpen = { ...accordionsOpen };
+    newAccordionsOpen[key] = !accordionsOpen[key];
+    setAccordionsOpen(newAccordionsOpen);
+  };
+  const onDetectorNameChange = (e, field) => {
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) => index < (aggList.length < 5 ? aggList.length : 5)
+  );
+
+  const getFeatureNameFromParams = (id) => {
+    let name = find(embeddable.vis.params.seriesParams, function (param) {
+      if (param.data.id === id) {
+        return true;
+      }
+    });
+
+    return name.data.label.replace(/[^a-zA-Z0-9-_]/g, '_');
+  };
+
+  const handleValidationAndSubmit = (formikProps) => {
+    if (!isEmpty(formikProps.errors)) {
+      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
+      notification.toasts.addDanger('One or more input fields is invalid');
+    } else {
+      handleSubmit(formikProps.values);
+    }
+  };
+
+  const handleSubmit = (values, formikHelpers) => {
+    try {
+      const detectorToCreate = formikValueToDetector(values);
+      dispatch(createDetector(detectorToCreate)).then(async (response) => {
+        dispatch(startDetector(response.response.id)).then(
+          (startDetectorResponse) => {
+            notification.toasts.addSuccess(`Detector created: ${values.name}`);
+          }
+        );
+        enum VisLayerTypes {
+          PointInTimeEvents = 'PointInTimeEvents',
+        }
+        const fn = {
+          type: VisLayerTypes.PointInTimeEvents,
+          name: 'overlay_anomalies',
+          args: {
+            detectorId: response.response.id,
+          },
+        } as VisLayerExpressionFn;
+
+        const savedObjectToCreate = {
+          title: embeddable.vis.title,
+          pluginResourceId: response.response.id,
+          visId: embeddable.vis.id,
+          savedObjectType: 'visualization',
+          visLayerExpressionFn: fn,
+        } as ISavedAugmentVis;
+
+        const savedObject = await createAugmentVisSavedObject(
+          savedObjectToCreate
+        );
+
+        const saveObjectResponse = await savedObject.save({});
+      });
+      closeFlyout();
+    } catch (e) {
+    } finally {
+      formikHelpers.setSubmitting(false);
+    }
+  };
+
+  function formikValueToDetector(values): Detector {
```

Comment:
You may use the existing helper fn here: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ReviewAndCreate/utils/helpers.ts#L29

Reply:
The feature list to UI metadata conversion is different than the existing helper function, cause when we first built the feature list, we built it based on the embeddable vis param series field, the format is slightly different than the existing feature lit.

Reply:
At the least, we could modify the helper fn to handle this case so we don't duplicate.